compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2346,19 +2413,34 @@ fn linker_with_args(
         .native_libraries
         .iter()
         .filter_map(|(&cnum, libraries)| {
-            (dependency_linkage[cnum] != Linkage::Static).then_some(libraries)
+            if sess.target.is_like_windows {
```

Comment:
This is fine as is for now, but once Apple targets support raw-dylib, this static deps skipping behavior should likely apply there too if you are allowed to put `.tbd` files in static libraries.